test/unit/api_spec.js

```diff
@@ -377,24 +403,28 @@ describe('api', function() {
     it('gets javascript with printing instructions (Print action)', function() {
       // PDF document with "Print" Named action in OpenAction
       var pdfUrl = combineUrl(window.location.href, '../pdfs/bug1001080.pdf');
-      var promise = PDFJS.getDocument(pdfUrl).then(function(doc) {
-        return doc.getJavaScript();
-      });
-      waitsForPromiseResolved(promise, function (data) {
-        expect(data).toEqual(['print({});']);
-        expect(data[0]).toMatch(viewerPrintRegExp);
+      var loadingTask = PDFJS.getDocument(pdfUrl);
+
+      loadingTask.promise.then(function (pdfDocument) {
+        pdfDocument.getJavaScript().then(function (javascript) {
+          expect(javascript).toEqual(['print({});']);
+          expect(javascript[0]).toMatch(viewerPrintRegExp);
+          loadingTask.destroy();
+        });
       });
     });
     it('gets javascript with printing instructions (JS action)', function() {
       // PDF document with "JavaScript" action in OpenAction
       var pdfUrl = combineUrl(window.location.href, '../pdfs/issue6106.pdf');
-      var promise = PDFJS.getDocument(pdfUrl).then(function(doc) {
-        return doc.getJavaScript();
-      });
-      waitsForPromiseResolved(promise, function (data) {
-        expect(data).toEqual(
-          ['this.print({bUI:true,bSilent:false,bShrinkToFit:true});']);
-        expect(data[0]).toMatch(viewerPrintRegExp);
+      var loadingTask = PDFJS.getDocument(pdfUrl);
+
+      loadingTask.promise.then(function (pdfDocument) {
```

Comment:
This probably needs to look similar to the `diff` above, in order to work correctly.